doc/api/cluster.markdown

```diff
@@ -2,12 +2,40 @@
 
     Stability: 2 - Stable
 
-A single instance of Node.js runs in a single thread. To take advantage of
-multi-core systems the user will sometimes want to launch a cluster of Node.js
-processes to handle the load.
+Node.js executes JavaScript code within the context of an event loop. Under the 
+hood, non-blocking socket I/O runs on the main event loop thread while blocking 
+DNS and filesystem I/O runs on a secondary thread that notifies the main thread 
+when it's finished. This concurrency model allows a blocking 
+[I/O bound](https://en.wikipedia.org/wiki/I/O_bound) application to easily 
+distribute work across multiple CPU cores. For more details, see the 
+[libuv](http://docs.libuv.org/en/v1.x/design.html) project.
+
+The `cluster` API allows a non-blocking I/O bound application to distribute 
+work across multiple CPU cores. A Node.js cluster is created when a **master 
+process** forks one or more **worker processes**. By design, workers use the 
+same application code as the master. Conveniently, the `cluster.isMaster` 
+property determines if a running process is the master or a worker.
 
-The cluster module allows you to easily create child processes that
-all share server ports.
+```js
+const cluster = require('cluster');
+if (cluster.isMaster) {
+  for (var i = 0; i < numCPUs; i++) {
```

Comment:
I generally prefer `let` in loops but I see both `var` or `let` in the docs. (https://github.com/nodejs/node/blob/master/doc/api/buffer.markdown has both) 
Excuse me if this has been brought up before.